bot/cogs/nasa.py

```diff
@@ -0,0 +1,166 @@
+            await ctx.send("No results found!")
+
+    @command(aliases=["nid"])
+    async def nasa_id(self, ctx: Context, id: str) -> None:
+        """Search for a picture on Nasa with nasa id."""
+        async with self.session.get(f"https://images-api.nasa.gov/asset/{id}") as resp:
+            data = await resp.json()
+
+        try:
+            items = data["collection"]["items"][0]["href"]
+            embed = Embed(color=Color.blurple())
+            embed.set_image(url=items)
+            embed.set_footer(text=f"ID: {id} | Powered by HotWired")
+        except KeyError:
+            await ctx.send("No results found!")
+
+        await ctx.send(embed=embed)
+
+    @command(aliases=["nasapatent"])
+    async def nasa_patent(self, ctx: Context, *, patent: str) -> None:
+        """Search for a picture on Nasa with nasa id."""
+        async with self.session.get(f"https://api.nasa.gov/techtransfer/patent/?{patent}&api_key={NASA_API}") as resp:
+            data = await resp.json()
+
+        items = data["results"]
+        if len(items) > 0:
+            rand_item = random.randint(0, len(items) - 1)
+            item = items[rand_item]
+
+            if len(remove_tags(item[3][:2000])) > 2048:
+                description = f"{item[3][:2000][:2045].strip()}..."
+            else:
+                description = item[3][:2000]
+
+            embed = Embed(
+                title=remove_tags(item[2]),
+                description=remove_tags(description),
+                color=Color.blurple()
+            )
+            embed.set_footer(text="Powered by HotWired")
+
+            await ctx.send(embed=embed)
+        else:
+            await ctx.send("No results Found!")
+
+    @command()
+    async def epic(self, ctx: Context, max: int = 1) -> None:
+        """Get images from DSCOVR's Earth Polychromatic Imaging Camera. can specify a maximum number of images to retrieve."""
+        async with self.session.get("https://epic.gsfc.nasa.gov/api/images.php") as response:
+            json = await response.json()
+
+            for i in range(min(max, len(json))):
+                embed = Embed(
+                    ittle="EPIC image",
+                    description=json[i].get("caption"),
+                    colour=Color.blurple()
+                )
+                embed.set_image(url="https://epic.gsfc.nasa.gov/epic-archive/jpg/" + json[i]["image"] + ".jpg")
+
+                await ctx.send(embed=embed)
+
+    @command()
+    async def mars(self, ctx: Context, date: str, rover: str = None, number: int = 1) -> None:
+        """Get images from Mars. must specify the date (in the form YYYY-MM-DD),and you can specify the rover and the number of images to retrieve."""
+        if not rover.lower() in ("curiosity", "opportunity", "spirit"):
+            await ctx.send("Sorry but this rover doesn't exist")
+            return
+
+        if rover is None:
+            rover = random.choice(["curiosity", "opportunity", "spirit"])
```

Comment:
This will first check if the rover is one of these, if not it will return (end function) so the default `None` will cause the function to return. The second if won't even have a chance to be called because of the first one.

You need to swap these, you need to first check if the rover is None and if it is, set one of these random options and only after that you can check if the `rover` is set correctly by the second if